test/e2e/provision.go

```diff
@@ -0,0 +1,90 @@
+package e2e
+
+import (
+	"context"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"os"
+	"path"
+	"sigs.k8s.io/e2e-framework/klient/decoder"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"time"
+)
+
+// CloudProvision defines operations to provision the environment on cloud providers.
+type CloudProvision interface {
+	CreateCluster(ctx context.Context, cfg *envconf.Config) error
+	CreateVPC(ctx context.Context, cfg *envconf.Config) error
```

Comment:
Looks we need DeleteVPC and DeleteCluster method in this interface.

Reply:
Hi @huoqifeng , added those functions to the interface on my last update to this PR.